sktime/forecasting/compose/_skforecast_reduce.py

```diff
@@ -0,0 +1,445 @@
+        """Clone the regressor and transformers."""
+        from sklearn.base import clone
+
+        self._regressor = clone(self.regressor)
+
+        if self.transformer_y:
+            self._transformer_y = clone(self.transformer_y)
+
+        if self.transformer_exog:
+            self._transformer_exog = clone(self.transformer_exog)
+
+    def _create_forecaster(self: "SkforecastAutoreg"):
+        """Create ``skforecast.ForecasterAutoreg.ForecasterAutoreg`` model."""
+        from skforecast.ForecasterAutoreg import ForecasterAutoreg
+
+        return ForecasterAutoreg(
+            self._regressor,
+            self.lags,
+            transformer_y=self._transformer_y,
+            transformer_exog=self._transformer_exog,
+            weight_func=self.weight_func,
+            differentiation=self.differentiation,
+            fit_kwargs=self.fit_kwargs,
+            binner_kwargs=self.binner_kwargs,
+        )
+
+    @staticmethod
+    def _coerce_column_names(X: Optional[pd.DataFrame]):
+        if X is None:
+            return None
+
+        return X.rename(columns=lambda column_name: str(column_name))
+
+    @staticmethod
+    def _coerce_int_to_range_index(df):
+        new_df = df.copy(deep=True)
+        start = new_df.index[0]
+        stop = new_df.index[-1] + 1
+        if len(new_df.index) == 1:
+            step = 1
+        else:
+            step = new_df.index[1] - start
+
+        new_index = pd.RangeIndex(start, stop, step)
+        np.testing.assert_array_equal(new_df.index, new_index)
+        new_df.index = new_index
+        return new_df
+
+    @staticmethod
+    def _coerce_period_to_datetime_index(df):
+        new_df = df.copy(deep=True)
+        period_freq = new_df.index.freq
+        new_df.index = new_df.index.to_timestamp(how="e").normalize()
+        new_df.index.freq = period_freq
+        return new_df
+
+    def _make_index_compatible(self, df, input_var):
+        if df is None:
+            return None
+
+        new_df = df
+        if not isinstance(df.index, (pd.RangeIndex, pd.DatetimeIndex)):
+            if isinstance(df.index, pd.PeriodIndex):
+                new_df = self._coerce_period_to_datetime_index(df)
+            elif pd.api.types.is_integer_dtype(df.index):
+                try:
+                    new_df = self._coerce_int_to_range_index(df)
+                except AssertionError:
+                    raise ValueError(
+                        f"Coercion of index of {input_var} from integer pd.Index to "
+                        "pd.RangeIndex failed. Please ensure that indexes are equally "
+                        "spaced apart."
+                    )
+            else:
+                raise ValueError(
+                    f"{input_var} must have one of the following index types: "
+                    "pd.RangeIndex, pd.DatetimeIndex, pd.PeriodIndex, pd.Index"
+                    f"(int dtype). Found index of type: {type(df.index)}"
+                )
+        return new_df
```

Comment:
This is coresponding to range or datetime indices, correct? To reduce nesting, which increases mccabe complexity, I suggest you make a positive check for those first and return `df` as it is. And then proceed to nested checks, without nesting this time.

For successful coercion case, use try-except-else or similar.